src/lib/y2storage/autoinst_proposal.rb

```diff
@@ -0,0 +1,184 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage/storage_manager"
+require "y2storage/disk_analyzer"
+require "y2storage/proposal"
+require "y2storage/exceptions"
+
+module Y2Storage
+  # Class to calculate a storage proposal for autoinstallation
+  #
+  # @example Example
+  #   profile = Yast::Profile.current["partitioning"]
+  #   proposal = Y2Storage::AutoinstProposal.new(partitioning)
+  #   proposal.proposed?            # => false
+  #   proposal.proposed_devicegraph # => nil
+  #   proposal.planned_devices      # => nil
+  #
+  #   proposal.propose              # => Performs the calculation
+  #
+  #   proposal.proposed?            # => true
+  #   proposal.proposed_devicegraph # => Proposed layout
+  class AutoinstProposal
+    include Yast::Logger
+
+    # @return [Hash] Partitioning layout from an AutoYaST profile
+    attr_reader :partitioning
+
+    # @return [Devicegraph] Initial device graph
+    attr_reader :initial_devicegraph
+
+    # Proposed layout of devices, nil if the proposal has not been
+    # calculated yet
+    # @return [Devicegraph]
+    attr_reader :proposed_devicegraph
+
+    alias_method :devices, :proposed_devicegraph
+
+    # Constructor
+    #
+    # @param partitioning [Array<Hash>] Partitioning schema from an AutoYaST profile
+    # @param devicegraph  [Devicegraph] starting point. If nil, then probed devicegraph
+    #   will be used
+    # @param disk_analyzer [DiskAnalyzer] if nil, a new one will be created
+    #   based in the initial devicegraph
+    def initialize(partitioning: [], devicegraph: nil, disk_analyzer: nil)
+      @partitioning = partitioning
+      @initial_devicegraph = devicegraph
```

Comment:
if `devicegraph` is nil, where it is assigned to probed one? I cannot find it in code.